### ISSUE TITLE
Android interop test should create channel in background thread.

### ISSUE THREAD

**Issue by madongfly** [grpc/grpc-java#577]
When the host is not an ip address, addressing resolving will cause NetworkOnMainThreadException.

**Comment by ejona86** [comment]
Channel creation should be fine. newCall is the issue.

We really should fix OkHttp transport's newCall to not be synchronous. It would fix multiple problems.

**Comment by madongfly** [comment]
I didn't get you, for this particular issue, the problem is that when the host is not an explicitly ip address, it would trigger: 
```
06-29 11:44:18.321: E/AndroidRuntime(2414): Caused by: android.os.NetworkOnMainThreadException
06-29 11:44:18.321: E/AndroidRuntime(2414): 	at android.os.StrictMode$AndroidBlockGuardPolicy.onNetwork(StrictMode.java:1145)
06-29 11:44:18.321: E/AndroidRuntime(2414): 	at java.net.InetAddress.lookupHostByName(InetAddress.java:385)
06-29 11:44:18.321: E/AndroidRuntime(2414): 	at java.net.InetAddress.getAllByNameImpl(InetAddress.java:236)
06-29 11:44:18.321: E/AndroidRuntime(2414): 	at java.net.InetAddress.getByName(InetAddress.java:289)
06-29 11:44:18.321: E/AndroidRuntime(2414): 	at java.net.InetSocketAddress.<init>(InetSocketAddress.java:105)
06-29 11:44:18.321: E/AndroidRuntime(2414): 	at java.net.InetSocketAddress.<init>(InetSocketAddress.java:90)
06-29 11:44:18.321: E/AndroidRuntime(2414): 	at io.grpc.transport.okhttp.OkHttpChannelBuilder.forAddress(OkHttpChannelBuilder.java:87)
06-29 11:44:18.321: E/AndroidRuntime(2414): 	at io.grpc.android.integrationtest.InteropTester.<init>(InteropTester.java:86)
06-29 11:44:18.321: E/AndroidRuntime(2414): 	at io.grpc.android.integrationtest.TesterInstrumentation.onCreate(TesterInstrumentation.java:86)
06-29 11:44:18.321: E/AndroidRuntime(2414): 	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:4319)
```

That's why it wasn't exposed when I tested it against the server running on GCE, but was exposed when I tried "grpc-test.sandbox.google.com".

I didn't see newCall is involved.

**Comment by ejona86** [comment]
Ah, I see. This will be solved by #579, then.

**Comment by madongfly** [comment]
Closing it since #579 is in.
